### PR TITLE
feat: implement habits part 1

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR

Added habit tracking functionality with interactive completion cards and confetti animations

I also had to do some refactoring with the API and types in order to avoid import cycles. I think it's better now anyways because now the types for habit stuff is in the `/api/habits` folder.

I'll make another PR for further implementation of the habit cards

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/zZJDTogQbjQWXEHwCYBl/f142dfe2-c265-4b25-832f-ec0e54cc67e0.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/zZJDTogQbjQWXEHwCYBl/f142dfe2-c265-4b25-832f-ec0e54cc67e0.mov">Screen Recording 2024-12-07 at 12.34.25 PM.mov</video>

### What changed?

- Introduced habit tracking data structures and API endpoints
- Created HabitCard component with weekly view completion tracking
- Added confetti animation system for habit completion celebrations
- Implemented habit color theming system
- Added new dependency lottie-react-native for confetti animation

### How to test?

1. Navigate to the habits tab
2. View the list of habits with their weekly completion status
3. Tap on a day's square to mark a habit as complete
4. Verify confetti animation plays when marking complete
5. Verify progress tracking updates correctly
6. Test dark/light mode color transitions

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/habits-together/app/137?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#137** <a href="https://app.graphite.dev/github/pr/habits-together/app/137?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/137?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#136** <a href="https://app.graphite.dev/github/pr/habits-together/app/136?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#135** <a href="https://app.graphite.dev/github/pr/habits-together/app/135?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#134** <a href="https://app.graphite.dev/github/pr/habits-together/app/134?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#133** <a href="https://app.graphite.dev/github/pr/habits-together/app/133?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#132** <a href="https://app.graphite.dev/github/pr/habits-together/app/132?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by AnkushSarkar10 — APPROVED** [review]

**Review by AnkushSarkar10 — APPROVED** [review]
